cli/azd/pkg/azdo/pipeline.go

```diff
@@ -0,0 +1,231 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package azdo
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/azure/azure-dev/cli/azd/pkg/environment"
+	"github.com/azure/azure-dev/cli/azd/pkg/infra/provisioning"
+	"github.com/azure/azure-dev/cli/azd/pkg/input"
+	"github.com/microsoft/azure-devops-go-api/azuredevops"
+	"github.com/microsoft/azure-devops-go-api/azuredevops/build"
+	"github.com/microsoft/azure-devops-go-api/azuredevops/taskagent"
+)
+
+// Creates a variable to be associated with a Pipeline
+func createBuildDefinitionVariable(value string, isSecret bool, allowOverride bool) build.BuildDefinitionVariable {
+	return build.BuildDefinitionVariable{
+		AllowOverride: &allowOverride,
+		IsSecret:      &isSecret,
+		Value:         &value,
+	}
+}
+
+// returns the default agent queue. This is used to associate a Pipeline with a default agent pool queue
+func getAgentQueue(ctx context.Context, projectId string, connection *azuredevops.Connection) (*taskagent.TaskAgentQueue, error) {
+	client, err := taskagent.NewClient(ctx, connection)
+	if err != nil {
+		return nil, err
+	}
+	getAgentQueuesArgs := taskagent.GetAgentQueuesArgs{
+		Project: &projectId,
+	}
+	queues, err := client.GetAgentQueues(ctx, getAgentQueuesArgs)
+	if err != nil {
+		return nil, err
+	}
+	for _, queue := range *queues {
+		if *queue.Name == "Default" {
+			return &queue, nil
+		}
+	}
+	return nil, fmt.Errorf("could not find a default agent queue in project %s", projectId)
+}
+
+// find pipeline by name
+func pipelineExists(
+	ctx context.Context,
+	client *build.Client,
+	projectId *string,
+	pipelineName *string,
+) (bool, error) {
+	getDefinitionsArgs := build.GetDefinitionsArgs{
+		Project: projectId,
+		Name:    pipelineName,
+	}
+
+	buildDefinitionsResponse, err := (*client).GetDefinitions(ctx, getDefinitionsArgs)
```

Comment:
Yes the client was implemented using values. We could pass the client value in instead of a pointer and it removes the need to get the value. 
https://github.com/kaizentm/azure-dev/blob/add-azdo-provider/cli/azd/pkg/azdo/pipeline.go#L51

Reply:
Done I modified it so the client is passed in by value and that removes the need to get the pointer value.